### PR TITLE
Add end to end support for comparison ops

### LINKED ISSUE #632: [Ops] Add TTIR comparison OPs
We should add `eq`, `ne`, `le`, `lt`, `ge`, `gt` OPs to TTIR and TTNN dialects, and map them to [`ttnn.eq`](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/eq.html), [`ttnn.ne`](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/ne.html), [`ttnn.le`](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/le.html), [`ttnn.lt`](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/lt.html), [`ttnn.ge`](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/ge.html), [`ttnn.gt`](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/gt.html) OPs.

### PR DESCRIPTION
**Pull request by mmanzoorTT**
* equal (eq)
* not equal (ne)
* less than or equal to (le)
* less than (lt)
* greater than (gt)

closes #632

### CODE REVIEW

**Review by nsmithtt — APPROVED** [review]

**Review by AleksKnezevic — APPROVED** [review]

File: test/ttmlir/Dialect/TTNN/simple_compare.mlir [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    // CHECK: %[[C:.*]] = "ttnn.empty"[[C:.*]]
+    %0 = tensor.empty() : tensor<13x31xf32>
+    // CHECK: %[[C:.*]] = "ttnn.ne"[[C:.*]]
+    %1 = "ttir.ne"(%arg0, %arg1, %0) <{operandSegmentSizes = array<i32: 2, 1>, operand_constraints = [#any_device, #any_device, #any_device]}> : (tensor<13x31xf32>, tensor<13x31xf32>, tensor<13x31xf32>) -> tensor<13x31xf32>
+    return %1 : tensor<13x31xf32>
+  }
+}
+
+module attributes {} {
+  func.func @forward(%arg0: tensor<64x128xf32>, %arg1: tensor<64x128xf32>) -> tensor<64x128xf32> {
```
Comment (sdjordjevicTT) [inline test/ttmlir/Dialect/TTNN/simple_compare.mlir:24]: Can you please rename this forward to greater_equal?

File: test/ttmlir/Silicon/TTNN/simple_compare.mlir [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  return %1 : tensor<13x31xf32>
+}
+
+func.func @greater_equal(%arg0: tensor<13x31xf32>, %arg1: tensor<13x31xf32>) -> tensor<13x31xf32> {
+  // CHECK: %[[C:.*]] = "ttnn.empty"[[C:.*]]
+  %0 = tensor.empty() : tensor<13x31xf32>
+  // CHECK: %[[C:.*]] = "ttnn.ge"[[C:.*]]
+  %1 = "ttir.ge"(%arg0, %arg1, %0) <{operandSegmentSizes = array<i32: 2, 1>, operand_constraints = [#any_device, #any_device, #any_device]}> : (tensor<13x31xf32>, tensor<13x31xf32>, tensor<13x31xf32>) -> tensor<13x31xf32>
+  return %1 : tensor<13x31xf32>
+}
```
Comment (sdjordjevicTT) [inline test/ttmlir/Silicon/TTNN/simple_compare.mlir:29]: nit: new line

File: test/ttmlir/Silicon/TTNN/simple_compare.mlir [hunk 3]
```diff
@@ -0,0 +1,52 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline="system-desc-path=%system_desc_path%" %s > %t.mlir
+// RUN: FileCheck %s --input-file=%t.mlir
+// RUN: ttmlir-translate --ttnn-to-flatbuffer %t.mlir > %t.ttnn
+#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
+#any_device_tile = #tt.operand_constraint<dram|l1|tile|any_device_tile>
+
+func.func @equal(%arg0: tensor<13x31xf32>, %arg1: tensor<13x31xf32>) -> tensor<13x31xf32> {
```
Comment (sdjordjevicTT) [inline test/ttmlir/Silicon/TTNN/simple_compare.mlir:7]: Just for my better understanding, we don't have module defined in this file, just functions?

**Review by sdjordjevicTT — APPROVED** [review]

File: test/ttmlir/Dialect/TTNN/simple_compare.mlir (hunk 1 above)
Comment (mmanzoorTT) [reply test/ttmlir/Dialect/TTNN/simple_compare.mlir]: Thanks for pointing it out. updated.

File: test/ttmlir/Silicon/TTNN/simple_compare.mlir (hunk 2 above)
Comment (mmanzoorTT) [reply test/ttmlir/Silicon/TTNN/simple_compare.mlir:29]: fixed. thanks

File: test/ttmlir/Silicon/TTNN/simple_compare.mlir (hunk 3 above)
Comment (mmanzoorTT) [reply test/ttmlir/Silicon/TTNN/simple_compare.mlir:7]: AFAIK module can be used to group functions together but it’s not required.

**Pull request merged**